packages/itwin/tree-widget/src/test/trees/models-tree/internal/ModelsTreeVisibilityHandler.test.ts

```diff
@@ -3235,6 +3237,159 @@ describe("ModelsTreeVisibilityHandler", () => {
       });
     });
 
+    describe("IsAlwaysDrawnExclusive is true", () => {
+      it("changing model visibility does not affect other models", async function () {
+        await using buildIModelResult = await buildIModel(this, async (builder) => {
+          const model1 = insertPhysicalModelWithPartition({ builder, codeValue: "1" }).id;
+          const category1 = insertSpatialCategory({ builder, codeValue: "category1" }).id;
+          const element1 = insertPhysicalElement({ builder, modelId: model1, categoryId: category1 }).id;
+          const element2 = insertPhysicalElement({ builder, modelId: model1, categoryId: category1 }).id;
+
+          const otherModel = insertPhysicalModelWithPartition({ builder, codeValue: "2" }).id;
+          const otherCategory = insertSpatialCategory({ builder, codeValue: "category2" }).id;
+          insertPhysicalElement({ builder, modelId: otherModel, categoryId: otherCategory });
+
+          return { model1, category1, element1, element2, otherModel, otherCategory };
+        });
+
+        const { imodel, ...ids } = buildIModelResult;
+        using visibilityTestData = createVisibilityTestData({ imodel });
+        const { handler, provider, viewport } = visibilityTestData;
+        viewport.changeModelDisplay(ids.model1, true);
+        viewport.setAlwaysDrawn(new Set([ids.element2]), true);
+        viewport.renderFrame();
+        await handler.changeVisibility(createModelHierarchyNode(ids.model1), true);
```

Comment:
maybe it would be worth checking hierarchy visibility state  before doing the change? just to confirm it matches the state we expect.